configure.ac

```diff
@@ -90,40 +110,49 @@ AM_COND_IF([DEPENDENCY_CHECKS], [
             ;;
     esac
 
-    PKG_CHECK_MODULES(HARFBUZZ, harfbuzz, [
-        PKG_CHECK_MODULES(FREETYPE, freetype2,[],[AC_MSG_FAILURE([Harfbuzz found but not freetype2])])
-        PKG_CHECK_MODULES(FONTCONFIG, fontconfig,[],[AC_MSG_FAILURE([Harfbuzz found but not fontconfig])])
-        with_harfbuzz=yes
-        backend=harfbuzz
-    ], [
-        if test "x$with_harfbuzz" != xcheck; then
-            AC_MSG_FAILURE([--with-harfbuzz was given, but test for harfbuzz failed])
-        fi
-    ])
+    AM_COND_IF([FONTCONFIG],
+               [PKG_CHECK_MODULES(FONTCONFIG, [fontconfig],
+                                  [],
+                                  [AC_MSG_FAILURE([Fontconfig package not found])])])
+
+    AM_COND_IF([HARFBUZZ],
+               [PKG_CHECK_MODULES(HARFBUZZ, [harfbuzz >= 2.7.4],
```

Comment:
> We were not requiring any specific version, but it is unlikely that the
> code works with such old versions of HarfBuzz. Instead, require the
> newest version of HarfBuzz we can get away with, the version available
> in stable Debian/Ubuntu.

So, this is turning out to be a little bit more complicated. Stable Debian/Ubuntu sounded like a good minimum baseline so I skipped over this. It might be true 2.7.4 is the minimum for Debian, but unfortunately for Ubuntu we have LTS releases out there with 1.x releases still in support. Bionic is on 1.7.2 and Focal is on 2.6.4.

I am not sure I want to revisit this since the most *recent* supported LTS will clear this bar, but it does mean we will be dropping support for Bionic and Focal.

I think I'm okay with that. I'm also looking at what it will take to backport HarfBuzz 6 back to at least Jammy so we can have full support everywhere we support Ubuntu at all, but I haven't found the bottom of the rabbit hole yet.

Reply:
No 6.x yet, but it looks like [this PPA](https://launchpad.net/~reviczky/+archive/ubuntu/harfbuzz/+packages) has at least 2.7.4 backported to all current LTS releases.